src/core/execute.c

```diff
@@ -2545,6 +2548,162 @@ static int write_credential(
         return 0;
 }
 
+static int load_credential(
+                const ExecContext *context,
+                const ExecParameters *params,
+                ExecLoadCredential *lc,
+                const char *unit,
+                int read_dfd,
+                int write_dfd,
+                uid_t uid,
+                bool ownership_ok,
+                uint64_t *left) {
+
+        assert(context);
+        assert(lc);
+        assert(unit);
+        assert(write_dfd);
```

Comment:
use `assert(write_dfd >= 0);` to check for validity of an fd, because 0 is a valid fd.